flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/TestBaseUtils.java

```diff
@@ -480,7 +480,8 @@ protected static File asFile(String path) {
 			}
 		}
 		
-		assertEquals("Wrong number of elements result", expectedStrings.length, resultStrings.length);
+		assertEquals(String.format("Wrong number of elements result. Expected: %s. Result: %s.", Arrays.toString(expectedStrings), Arrays.toString(resultStrings)),
```

Comment:
What if we moved `String.format` to its own line, included in the string both the array lengths and contents, and added a comment to describe why we are also printing the full arrays?

Also, should the arrays be printed on new lines such that they would line up until the diverging element? We'll need to move the sorting of the arrays before the length check.